comp/core/tagger/tagger.go

```diff
@@ -324,11 +327,48 @@ func (t *TaggerClient) ResetCaptureTagger() {
 // NOTE(remy): it is not needed to sort/dedup the tags anymore since after the
 // enrichment, the metric and its tags is sent to the context key generator, which
 // is taking care of deduping the tags while generating the context key.
-func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, udsOrigin string, clientOrigin string, cardinalityName string) {
-	cardinality := taggerCardinality(cardinalityName)
+func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
+	cardinality := taggerCardinality(originInfo.Cardinality)
+
+	switch originInfo.ProductOrigin {
+	case taggertypes.ProductOriginDogStatsD:
+		if *originInfo.OptOutEnabled && originInfo.Cardinality == "none" {
```

Comment:
```suggestion
		if originInfo.OptOutEnabled != nil && *originInfo.OptOutEnabled && originInfo.Cardinality == "none" {
```
I think it's better to be safe if we keep the pointer

Reply:
Yes that's done 👍